mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -1357,15 +1357,20 @@ class WidgetAdapter(
                 return false
             }
 
-            // hide dividers before and after frame widgets
+            // Hide dividers before and after frame widgets
             val adapter = parent.adapter
-            if (adapter != null) {
+            val viewTypeCurrentPos = adapter?.getItemViewType(position)
+            if (viewTypeCurrentPos != null) {
                 if (adapter.getItemViewType(position) == TYPE_FRAME) {
                     return true
                 }
                 if (position < adapter.itemCount - 1) {
                     if (adapter.getItemViewType(position + 1) in intArrayOf(TYPE_FRAME, TYPE_INVISIBLE)) {
                         return true
+                    } else if (viewTypeCurrentPos == TYPE_INVISIBLE &&
+                        adapter.getItemViewType(position - 1) == TYPE_FRAME
```

Comment:
This needs to be put into a separate if block with condition `position > 0`, so the check matches the used indices.

Reply:
Done